host/volume/api/http.go

```diff
@@ -26,6 +31,10 @@ func (api *HTTPAPI) RegisterRoutes(r *httprouter.Router) {
 	r.GET("/storage/volumes", api.List)
 	r.GET("/storage/volumes/:volume_id", api.Inspect)
 	r.PUT("/storage/volumes/:volume_id/snapshot", api.Snapshot)
+	// takes host and volID parameters, triggers a send on the remote host and give it a list of snaps already here, and pipes it into recv
+	r.POST("/storage/volumes/:volume_id/pullSnapshot", api.Pull)
+	// responds with a snapshot stream binary.  only works on snapshots, takes 'haves' parameters, usually called by a node that's servicing a 'pullSnapshot' request
+	r.POST("/storage/volumes/:volume_id/send", api.Send)
```

Comment:
It does feel like a GET operation, but there's a request body to hold the "haves".

`Content-Type: application/zfs-snapshot-stream`?  (I haven't found any other examples in the wild yet.)